test/test_markets.js

```diff
@@ -0,0 +1,94 @@
+const utils = require('./utils')
+
+const Event = artifacts.require('Event')
+const EventFactory = artifacts.require('EventFactory')
+const Token = artifacts.require('Token')
+const EtherToken = artifacts.require('EtherToken')
+const CentralizedOracle = artifacts.require('CentralizedOracle')
+const CentralizedOracleFactory = artifacts.require('CentralizedOracleFactory')
+const Market = artifacts.require('Market')
+const StandardMarketFactory = artifacts.require('StandardMarketFactory')
+const LMSRMarketMaker = artifacts.require('LMSRMarketMaker')
+
+contract('markets', function (accounts) {
+    let centralizedOracleFactory
+    let eventFactory
+    let etherToken
+    let standardMarketFactory
+    let lmsrMarketMaker
+
+    beforeEach(async () => {
+        centralizedOracleFactory = await CentralizedOracleFactory.deployed()
+        eventFactory = await EventFactory.deployed()
+        etherToken = await EtherToken.deployed()
+        standardMarketFactory = await StandardMarketFactory.deployed()
+        lmsrMarketMaker = await LMSRMarketMaker.deployed()
+    })
+
+    it('should buy and sell all outcomes', async () => {
+        // create event
+        const ipfsHash = 'QmYwAPJzv5CZsnA625s3Xf2nemtYgPpHdWEz79ojWnPbdG'
+        const oracle = utils.getParamFromTxEvent(
```

Comment:
On tricky thing is getting results for things that cause a state change. Basically you have to look at events that are fired during the transaction to try and get the return value.